src/renderer/VideoButton.tsx

```diff
@@ -102,14 +109,105 @@ export default function VideoButton(props: any) {
     handleCloseMenu();
   };
 
+  /**
+   * Seek the selected video to the specified relative timestamp
+   */
+  const seekVideo = (videoIndex: number, ts: number) => {
+    ipc.sendMessage('contextMenu', ['seekVideo', videoIndex, ts])
+    handleCloseMenu();
+  };
+
+  const getDungeonEncounterById = (id: number): string => {
+    if (dungeonEncounters.hasOwnProperty(id)) {
+      return dungeonEncounters[id]
+    }
+
+    return 'Unknown boss';
+  }
+
+  const buttonClasses = ['videoButton'];
+  const keystoneVideoSegments = []
+  let keystonePlusses;
+
+  if (isMythicPlus) {
+    buttonClasses.push('dungeon')
+
+    // If the video metadata doesn't contain its own alloted time array, we'll use whichever
+    // is current in the constants.
+    const keystoneAllottedTime = video.challengeMode.allottedTime ?? dungeonTimersByMapId[video.challengeMode.mapId];
+
+    // Calculate the upgrade levels of the keystone
+    const keystoneResult = calculateKeystoneCompletionResult(
+      keystoneAllottedTime,
+      video.challengeMode.duration
+    );
+    keystonePlusses = '+'.repeat(keystoneResult)
+
+    /**
+     * Generate the JSX for the video segments that are used in the context menu on
+     * the VideoButton.
+     */
+    const videoSegments = video.challengeMode.videoSegments.map((segment: any) => {
+      let videoSegmentMenu;
+      let segmentDurationText
```

Comment:
semi colons on a few lines here and below

Reply:
I have fixed it better and utilized the fact that `.push()` can take an arbitrary number of elements.

Reply:
I have fixed it better and utilized the fact that `.push()` can take an arbitrary number of elements.